scripts/extractSlideData.js

```diff
@@ -3,44 +3,78 @@ const path = require('path');
 
 const listFilesInDirectory = (directory, fileNames) => {
   const files = fs.readdirSync(directory);
-  files.forEach(file => {
+  files.forEach((file) => {
     const filePath = path.join(directory, file);
     if (fs.statSync(filePath).isDirectory()) {
       listFilesInDirectory(filePath, fileNames);
     } else {
       fileNames.push(filePath);
     }
   });
-}
+};
+
+const extractTitle = (path) => {
+  try {
+    const data = fs.readFileSync(path, 'utf8');
+    const titleRegex = /<title>(.*?)<\/title>/;
+    const match = data.match(titleRegex);
+
+    if (match) {
+      const titleContent = match[1];
+      return titleContent;
+    } else {
+      return '';
+    }
+  } catch (err) {
+    return err.message;
+  }
+};
 
 const extractSlideData = async () => {
-  const talksPath = path.resolve("../talks/");
-  const dataPath = path.resolve("../data/slides.json")
+  const talksPath = path.resolve('talks/');
+  const dataPath = path.resolve('data/slides.json');
   const htmlFilter = /\.html$/;
   const folderFilter = /\./;
 
-    fs.readdir(talksPath, (err, files) => {
-      try {
-        let talks = files.filter(file => {
-          if(!folderFilter.test(file)) {
-            const fileNames = [];
-            listFilesInDirectory(talksPath + "/" + file, fileNames);
+  fs.readdir(talksPath, (err, files) => {
```

Comment:
Can we use the `fs.readdirSync` method instead? So we have everything inside one try catch instead of one async (readdir) and other sync (existsSync, writeFileSync) methods?